include/dlaf/auxiliary/norm/mc/norm_max_L.h

```diff
@@ -0,0 +1,97 @@
+
+#include "dlaf/common/range2d.h"
+#include "dlaf/common/vector.h"
+#include "dlaf/communication/communicator_grid.h"
+#include "dlaf/communication/sync/reduce.h"
+#include "dlaf/lapack_tile.h"
+#include "dlaf/matrix.h"
+#include "dlaf/matrix/distribution.h"
+#include "dlaf/types.h"
+
+namespace dlaf {
+namespace internal {
+namespace mc {
+
+// Compute max norm of the lower triangular part of the distributed matrix
+// https://en.wikipedia.org/wiki/Matrix_norm#Max_norm
+//
+// It just addresses tiles with elements belonging to the lower triangular part of the matrix
+//
+// Thanks to the nature of the max norm, it is valid for:
+// - ge/sy/he lower
+// - tr lower non-unit
+template <class T>
+dlaf::BaseType<T> norm_max_L(comm::CommunicatorGrid comm_grid, Matrix<const T, Device::CPU>& matrix) {
+  using dlaf::common::internal::vector;
+  using dlaf::common::make_data;
+  // using dlaf::common::iterate_range2d;
+  using hpx::util::unwrapping;
+
+  using dlaf::tile::lange;
+  using dlaf::tile::lantr;
+
+  using NormT = dlaf::BaseType<T>;
+
+  const auto& distribution = matrix.distribution();
+
+  DLAF_ASSERT_SIZE_SQUARE(matrix);
+  DLAF_ASSERT_BLOCKSIZE_SQUARE(matrix);
+
+  if (GlobalElementSize{0, 0} == matrix.size())
+    return {0};
+
+  vector<hpx::future<NormT>> tiles_max;
+  tiles_max.reserve(distribution.localNrTiles().rows() * distribution.localNrTiles().cols());
+
+  // for each local tile in the (global) lower triangular matrix, create a task that finds the max element in the tile
+  for (auto tile_wrt_local : iterate_range2d(distribution.localNrTiles())) {
+    auto tile_wrt_global = distribution.globalTileIndex(tile_wrt_local);
+
+    if (tile_wrt_global.row() < tile_wrt_global.col())
+      continue;
+
+    bool is_diag = tile_wrt_global.row() == tile_wrt_global.col();
+    auto norm_max_f = unwrapping([is_diag](auto&& tile) noexcept->NormT {
+      if (is_diag)
+        return lantr(lapack::Norm::Max, blas::Uplo::Lower, blas::Diag::NonUnit, tile);
+      else
+        return lange(lapack::Norm::Max, tile);
+    });
+    auto current_tile_max = hpx::dataflow(norm_max_f, matrix.read(tile_wrt_local));
+
+    tiles_max.emplace_back(std::move(current_tile_max));
+  }
+
+  // than it is necessary to reduce max values from all ranks into a single max value for the matrix
+
+  // TODO unwrapping can be skipped for optimization reasons
+  NormT local_max_value = hpx::dataflow(unwrapping([](const auto&& values) {
+                                          if (values.size() == 0)
+                                            return std::numeric_limits<NormT>::min();
+                                          return *std::max_element(values.begin(), values.end());
+                                        }),
+                                        tiles_max)
+                              .get();
+  NormT max_value;
+  dlaf::comm::sync::reduce(0, comm_grid.fullCommunicator(), MPI_MAX, make_data(&local_max_value, 1),
```

Comment:
I'm wondering if it would make sense to expose as parameter where the norm result will be computed (i.e. the root rank of the reduce).

What do you think @rasolca @teonnik ?

Reply:
Yes, I think that makes sense. It will also be nice to be able to choose the result to be available on all processes (i.e. MPI_Allreduce).

Reply:
@rasolca @teonnik 

Do we want to proceed in this PR or do we want to address this later? In case I will open an issue, so we can also discuss how to change the API.

Reply:
I think it's OK if we address MPI_Allreduce in a separate PR, just adding the rank on which we want the result to the parameter list of the function and propagating it to `dlaf::comm::sync::reduce` should be sufficient for now.

Reply:
I've just added the parameter 😉